test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -474,3 +474,90 @@ UCS_TEST_SKIP_COND_P(test_ucp_peer_failure, disable_sync_send,
         EXPECT_EQ(UCS_ERR_UNSUPPORTED, UCS_PTR_STATUS(req));
     }
 }
+
+class test_ucp_peer_failure_keepalive : public test_ucp_peer_failure
+{
+public:
+    void init() {
+        test_ucp_peer_failure::init();
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    void kill_receiver()
+    {
+        ucp_ep_h ep = receiver().revoke_ep(0, 0);
+
+        void *creq = ucp_ep_close_nb(ep, UCP_EP_CLOSE_MODE_FORCE);
+        request_wait(creq);
+    }
+
+    void sendrecv()
+    {
+        char send_buf[256];
+        char recv_buf[256];
+
+        ucp_request_param_t send_param; 
+        ucp_request_param_t recv_param; 
+
+        send_param.op_attr_mask = 0;
+        recv_param.op_attr_mask = 0;
+
+        void *recv_req = ucp_tag_recv_nbx(receiver().worker(), recv_buf,
+                                          sizeof(recv_buf),
+                                          0, 0, &recv_param);
+        void *send_req = ucp_tag_send_nbx(sender().ep(), send_buf,
+                                          sizeof(send_buf),
+                                          0, &send_param);
+        request_wait(recv_req);
+        request_wait(send_req);
+    }
```

Comment:
why not use `test_ucp_peer_failure::smoke_test`?

Reply:
because used different entities to connect

Reply:
same as smoke_test

Reply:
reused smoke test